integration-tests/testkit/test.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * This module uses Vitest's fixture system to make common usage patterns
+ * of our testkit easily consumable in test cases. @see https://vitest.dev/guide/test-context.html#test-extend
+ */
+
+import { test as testBase } from 'vitest';
+import { CLI, createCLI } from './cli';
+import { ProjectType } from './gql/graphql';
+import { initSeed, OrgSeed, OwnerSeed, ProjectSeed, Seed, TargetAccessTokenSeed } from './seed';
+
+interface Context {
+  seed: Seed;
+  owner: OwnerSeed;
+  org: OrgSeed;
+  //
+  // "single" branch
+  //
+  projectSingle: ProjectSeed;
+  targetAccessTokenSingle: TargetAccessTokenSeed;
+  cliSingle: CLI;
+  //
+  // "federation" branch
+  //
+  projectFederation: ProjectSeed;
+  targetAccessTokenFederation: TargetAccessTokenSeed;
+  cliFederation: CLI;
+  //
+  // "stitching" branch
+  //
+  projectStitching: ProjectSeed;
+  targetAccessTokenStitching: TargetAccessTokenSeed;
+  cliStitching: CLI;
+}
+
+export const test = testBase.extend<Context>({
+  seed: async ({}, use) => {
+    const seed = await initSeed();
+    await use(seed);
```

Comment:
The `use` function tells me nothing, I use this function and I have no idea what it does. I need to dig in and it increases complexity of the setup by 10x.

Reply:
Vitest fixtures are powerful and worth learning, and you only need to learn it once. If you didn't learn it, it makes sense you don't know what it does. It's unfortunate that there's no JSDoc there. The gist of `use` is that it makes the given value available to a test e.g. `test('...', ({ seed }) => ...`.

Reply:
To me, this:

```
test(..., async () => {
  const seed = seedProject()
})
```

is clear to everyone that will ever see the integration test. They can click on it, find it easily, and if we ever migrate from `vitest` to `jest`, we won't have to move this extra thing that is very specific to vitest.

Your solution means that something in the test case context is added magically and we need to explain it to every person that will ever encounter it.
If that person is not aware of this vitest feature, searching for `seed` will mean 120 search results with that keyword and somewhere in those results, there's will be a correct answer.

Reply:
Personally I would advocate for effective use of the tools we use. I wasn't aware of any plans to switch to Jest, is that what you're implying (confused because my understanding up until now was the opposite, moving away from Jest)? If your point is just that it is specific to Vitest, that's only partly correct. Playwright test framework also has a similar system in core (while Node test does not). 

Anyways I don't have more points to offer than here https://github.com/graphql-hive/console/pull/6122#discussion_r1905441725. I'll remove the use of fixtures so we can move on to other topics in this PR.